src/main/java/io/prestosql/plugin/tiledb/TileDBPageSink.java

```diff
@@ -386,8 +392,11 @@ else if (isVarcharType(type) || isCharType(type)) {
         else if (VARBINARY.equals(type)) {
             columnBuffer.setItem(bufferPosition, type.getSlice(block, position).getBytes());
         }
-        else if (DATE.equals(type)) { // NOTE: this is not used because we make all date columns a varchar type
-            columnBuffer.setItem(bufferPosition, DATE_FORMATTER.print(TimeUnit.DAYS.toMillis(type.getLong(block, position))));
+        else if (DATE.equals(type)) {
+            columnBuffer.setItem(bufferPosition, type.getLong(block, position));
```

Comment:
You can't just set the date/timestamp directly. You have to do conversions to the real tiledb datatype. I.e. if the field is `TILEDB_DATETIME_YEAR` and you set it to a date it'll be stored wrong. I.e. if the user inserts `2020-01-1`, the expected value to be stored is only `50`.